clang/test/Driver/aarch64-ptrauth.c

```diff
@@ -1,5 +1,37 @@
+// RUN: %clang -### -c --target=aarch64 %s 2>&1 | FileCheck %s --check-prefix NONE
+// NONE: "-cc1"
+// NONE-NOT: "-fptrauth-
+
 // RUN: %clang -### -c --target=aarch64 -fno-ptrauth-intrinsics -fptrauth-intrinsics %s 2>&1 | FileCheck %s --check-prefix=INTRIN
 // INTRIN: "-cc1"{{.*}} "-fptrauth-intrinsics"
 
-// RUN: not %clang -### -c --target=x86_64 -fptrauth-intrinsics %s 2>&1 | FileCheck %s --check-prefix=ERR
-// ERR: error: unsupported option '-fptrauth-intrinsics' for target '{{.*}}'
+// RUN: %clang -### -c --target=aarch64 -fno-ptrauth-calls -fptrauth-calls %s 2>&1 | FileCheck %s --check-prefix=CALL
```

Comment:
The overhead of additional RUN lines perhaps overweigh 
the additional value testing each `-fptrauth-*` individually. Perhaps pack more on the same `%clang` RUN line?

Reply:
I've measured timings on my machine, and we can potentially save about 0.1s if combine some of the RUN lines. It's noticeable, but there are tests like aarch64-fixed-x-register.c which contain a lot more similar RUN lines which can also be combined and we could save even more if enhance them.

I suggest to merge this PR "as is" and probably submitting a subsequent PR which makes this test and some similar ones "lighter" if it's a big concern. As for me, keeping all the RUN lines can protect us from dummy typos and they probably are worth the time spent on them - but I don't mind combining the RUN lines if the extra overhead bothers many people.

Please let me know if this approach is OK for you and I can merge the PR. If the clang driver test timings look too bad, I can also submit a corresponding issue so the context of the discussion is not lost.

Reply:
I think there is some value separating tests for codegen but the value is largely diminished for driver.
For driver, the code looks like ` Args.addOptInFlag(CmdArgs, options::OPT_fptrauth_returns,
                    options::OPT_fno_ptrauth_returns);`. We don't repeat `OPT_fptrauth_returns` twice, so it's very difficult to make a typo mistake...

So for driver, I'd hope that some grouping is made. If that's extra trouble to you, you can submit this as is and I will simplify it a bit.

Reply:
OK, this makes sense. Given that, I've combined 7 RUN lines for individual options into a single RUN line (so the file contains 3 RUN lines total now). See b0f9a1985846448d01af9bacdb6e3ddb23d4588f.